### PR TITLE
Add eth_getBlockTransactionCountByNumber

### PR DESCRIPTION
**Pull request by mrwillis**
# Description

Added https://eth.wiki/json-rpc/API#eth_getblocktransactioncountbynumber

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have tested this code
- [ ] I have updated the README and other relevant documents (guides...)
- [ ] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

**Review by lazartravica — APPROVED** [review]

File: jsonrpc/eth_endpoint.go [hunk 1]
```diff
@@ -55,6 +55,31 @@ func (e *Eth) GetBlockByHash(hash types.Hash, fullTx bool) (interface{}, error)
...
+
+	case PendingBlockNumber:
+		return nil, fmt.Errorf("fetching the pending header is not supported")
+
+	default:
+		if number < 0 {
+			return nil, fmt.Errorf("invalid argument 0: block number larger than int64")
+		}
+		num = uint64(number)
+	}
```
Comment (Kourin1996) [inline jsonrpc/eth_endpoint.go]: I guess this part is the same to the one in `GetBlockByNumber`. Could I trouble you to define new function and move this?

**Review by Kourin1996 — COMMENTED** [review]
Thank you for the PR! It works locally. I have left a small comment just for code improvement. I'd appreciated if you could check. Thank you!

File: jsonrpc/eth_endpoint.go (hunk 1 above)
Comment (mrwillis) [reply jsonrpc/eth_endpoint.go]: Done

**Comment by mrwillis** [comment]
@Kourin1996 Is it possible to support the `pending` block parameter? I notice this is not supported but part of the spec 

https://eth.wiki/json-rpc/API#the-default-block-parameter

I assume `pending` in this context actually means blocks that aren't yet mined?

**Comment by Kourin1996** [comment]
> @Kourin1996 Is it possible to support the pending block parameter? I notice this is not supported but part of the spec
> https://eth.wiki/json-rpc/API#the-default-block-parameter
> I assume pending in this context actually means blocks that aren't yet mined?

Hello, @mrwillis. Thank you for your suggestions! I think so, `pending` means a block which is created but not completed mining yet. In IBFT, there is no such block. (But there is a block which has been produced but not confirmed)
We're starting discussion about this. And we can do this task in different PR.

**Comment by mrwillis** [comment]
> > @Kourin1996 Is it possible to support the pending block parameter? I notice this is not supported but part of the spec
> > https://eth.wiki/json-rpc/API#the-default-block-parameter
> > I assume pending in this context actually means blocks that aren't yet mined?
> 
> Hello, @mrwillis. Thank you for your suggestions! I think so, `pending` means a block which is created but not completed mining yet. In IBFT, there is no such block. (But there is a block which has been produced but not confirmed)
> We're starting discussion about this. And we can do this task in different PR.

Thanks @Kourin1996! Sounds good.

**Comment by zivkovicmilos** [comment]
> 
> 
> @Kourin1996 Is it possible to support the `pending` block parameter? I notice this is not supported but part of the spec
> 
> https://eth.wiki/json-rpc/API#the-default-block-parameter
> 
> I assume `pending` in this context actually means blocks that aren't yet mined?

Hey @mrwillis,

Currently we don't have any support for `pending` or I think `earliest`. 
Would this be something that you'd need soon?

**Review by zivkovicmilos — APPROVED** [review]
Looks good to me, thank you for the contribution 🙏 💯

**Comment by mrwillis** [comment]
> > @Kourin1996 Is it possible to support the `pending` block parameter? I notice this is not supported but part of the spec
> > https://eth.wiki/json-rpc/API#the-default-block-parameter
> > I assume `pending` in this context actually means blocks that aren't yet mined?
> 
> Hey @mrwillis,
> 
> Currently we don't have any support for `pending` or I think `earliest`. Would this be something that you'd need soon?

Hey @zivkovicmilos, so I only ask for this as it seems to be used by block explorers to see the transactions in the "pending" block

**Review by Kourin1996 — APPROVED** [review]
Approved. Sorry actually I checked latest PR before but I didn't change status.

**Pull request merged**